pkg/types/nutanix/helpers.go

```diff
@@ -194,3 +196,94 @@ func CategoryKey(infraID string) string {
 	categoryKey := fmt.Sprintf("%s%s", categoryKeyPrefix, infraID)
 	return categoryKey
 }
+
+// GetGPUList returns a list of VMGpus for the given list of GPU identifiers in the Prism Element (uuid).
+func GetGPUList(ctx context.Context, client *nutanixclientv3.Client, gpus []machinev1.NutanixGPU, peUUID string) ([]*nutanixclientv3.VMGpu, error) {
+	vmGPUs := make([]*nutanixclientv3.VMGpu, 0)
+
+	if len(gpus) == 0 {
+		return vmGPUs, nil
+	}
+
+	peGPUs, err := GetGPUsForPE(ctx, client, peUUID)
+	if err != nil {
+		return nil, err
```

Comment:
can you wrap this error(and other unwrapped errors) to give clarity on the context of the error?

Reply:
fixed.